bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,436 @@
+ */
+public final class Interpreter implements AutoCloseable {
+  private static final String TAG = Interpreter.class.getCanonicalName();
+
+  /**
+   * Options class for controlling runtime interpreter behavior.
+   *
+   * <p>Currently only {@link #setNumThreads(int)} is supported.
+   */
+  public static class Options {
+    int numThreads = -1;
+
+    public Options() {}
+
+    /**
+     * Sets the number of threads to be used for ops that support multi-threading.
+     *
+     * <p>{@code numThreads} should be >= -1. Setting {@code numThreads} to 0 has the effect to
+     * disable multithreading, which is equivalent to setting {@code numThreads} to 1. If
+     * unspecified, or set to the value -1, the number of threads used will be
+     * implementation-defined and platform-dependent.
+     */
+    public Options setNumThreads(int numThreads) {
+      this.numThreads = numThreads;
+      return this;
+    }
+  }
+
+  private final Tensor[] inputTensors;
+  private final Tensor[] outputTensors;
+  private final long nativeAddress;
+
+  private long inferenceDurationNanoseconds;
+  private boolean tensorsAllocated;
+
+  /**
+   * Initializes the Interpreter with model and options.
+   *
+   * @param modelByteBuffer: a directly allocated, native {@link java.nio.ByteOrder} byte buffer of
+   *     an IREE compatible TFLite model.
+   * @param options: options for the interpreter, or null (to use defaults).
+   * @throws IllegalArgumentException if the model cannot be initialized in the Interpreter.
+   */
+  public Interpreter(@NonNull ByteBuffer modelByteBuffer, Options options) throws Exception {
+    TensorFlowLite.init();
+    if (options == null) {
+      options = new Options();
+      options.setNumThreads(2);
+    }
+    nativeAddress = nativeNew(modelByteBuffer, options.numThreads);
+
+    if (nativeAddress == 0) {
+      throw new IllegalArgumentException("Could not create Interpreter");
+    }
+
+    inputTensors = new Tensor[getInputTensorCount()];
+    outputTensors = new Tensor[getOutputTensorCount()];
+  }
+
+  /**
+   * Runs model inference with a single input/output pair.
+   *
+   * <p>This API is compatible with the following buffer types:
+   *
+   * <ul>
+   *   <li>{@link ByteBuffer} - compatible with any underlying primitive Tensor type.
+   *   <li>{@link java.nio.FloatBuffer} - compatible with float Tensors.
+   *   <li>{@link java.nio.IntBuffer} - compatible with int32 Tensors.
+   *   <li>{@link java.nio.LongBuffer} - compatible with int64 Tensors.
+   * </ul>
+   *
+   * String input is currently not supported.
+   *
+   * <p>The API is most efficient with direct byte buffers. All buffers must be in native {@link
+   * java.nio.ByteOrder}.
+   *
+   * <p>Scalar input should be wrapped in the appropriate buffer of capacity 1. Ex: {@code
+   * inputFloatBuffer.put(1.0f);}.
+   *
+   * <pre>{@code
+   * FloatBuffer inputBuffer = ... create float buffer of capacity 1 ...
+   *
+   * }</pre>
```

Comment:
just curious - any reason for the extra newline in this code block?

```suggestion
   * <pre>{@code
   * FloatBuffer inputBuffer = ... create float buffer of capacity 1 ...
   * }</pre>
```